### ISSUE TITLE
Develop branch role

### ISSUE THREAD

**Issue by russel** [intellij-dlanguage/intellij-dlanguage#213]
I am getting confused about the roles of master and develop branches. We do not seem to have the "master for work, named branches for maintenance" nor the "master for releases develop for work" approaches. I have to admit I am more used to the former. Currently though master and develop seem to be in the "for work" position.

**Comment by SingingBush** [comment]
develop is always in state of flux. I am planning to write up a CONTRIBUTING.md with to specify the roles of branches. It got a little messy as there were problem with the auto deploy process from travis-ci. That's sorted now so I can give definitive answer.

**DEVELOP**: all pull requests should be made to this branch (it's used for merging and very minor changes). It's worth making local _feature_ & _bugfix_ branches when working on the code base then making a PR from them directly into this repos DEVELOP branch.

**MASTER**: merging into this branch will kick of a release to the EAP channel in Jetbrains repo. So I generally do some manual testing of the changes in develop before and make sure things look right before merging changes into master.

**RELEASE** (_can be ignored_): this is new and purely for releasing to the stable channel of Jetbrains repo (I had been trying to get travis-ci to do this when I create a TAG but for some reason it ignored the last tag so I created this branch)

also there's a ton of old branches. I'm not sure if there's anything useful in them and which have been merged but it maybe worth removing them.
